### PR TITLE
log: Add missing z,j, & t types to the logger

### PR DESCRIPTION
**Pull request by chrissie-c**
For problem #215

### CODE REVIEW

File: lib/log_format.c [hunk 1]
```diff
@@ -536,6 +536,33 @@ qb_vsnprintf_serialize(char *serialize, size_t max_len,
...
+			format++;
+			if (sizeof(size_t) == sizeof(long long)) {
+				type_longlong = QB_TRUE;
+			}
+			else {
+				type_long = QB_TRUE;
+			}
+			goto reprocess;
+		case 't':
+ 			format++;
```
Comment (jnpkrn) [inline lib/log_format.c]: Please, do not mix spaces and tabs.
`git diff` (or perhaps `git diff --color`) should expose that visually
(using `color.diff.whitespace`) by default.

File: lib/log_format.c [hunk 2]
```diff
@@ -536,6 +536,33 @@ qb_vsnprintf_serialize(char *serialize, size_t max_len,
 				format++;
 			}
 			goto reprocess;
+		case 'z':
+			format++;
+			if (sizeof(size_t) == sizeof(long long)) {
+				type_longlong = QB_TRUE;
+			}
+			else {
```
Comment (jnpkrn) [inline lib/log_format.c]: This is not common style/indenting for follow-up `else` statements in libqb.

**Comment by chrissie-c** [comment]
TBH the space/tab thing is a bit random all over even that file, but I've tidied it up and fixed the else format.

File: lib/log_format.c [hunk 3]
```diff
@@ -737,6 +761,35 @@ qb_vsnprintf_deserialize(char *string, size_t str_len, const char *buf)
 				type_longlong = QB_TRUE;
 			}
 			goto reprocess;
+		case 'z':
+			fmt[fmt_pos++] = *format;
+			format++;
+			if (sizeof(size_t) == sizeof(long long)) {
+				type_long = QB_FALSE;
```
Comment (jnpkrn) [inline lib/log_format.c]: `QB_FALSE` assignments for `'z'` are superfluous (unlike in `'l'` case).
At least if we don't want to care about behavior that is undefined anyway :-)

**Comment by jnpkrn** [comment]
There's one more cosmetic discrepancy.
Isn't it better to `--amend`/merge and `git push --force` to keep the history compact?

**Comment by chrissie-c** [comment]
git merging makes me break out in a cold sweat.

**Comment by kgaillot** [comment]
This looks good to me, and is a good idea.

Unfortunately, it doesn't really help us in pacemaker, since we have to stay compatible with older libqb versions. We'll have to take all these formatters out and cast to long long instead. (That seems better than defining some ugly constants for format and cast depending on libqb support.)

**Comment by chrissie-c** [comment]
Added as commit 22ac41108ec3ad273905c67ac5ef5be01bb0f268

**Comment by chrissie-c** [comment]
On 17/06/16 22:22, Ken Gaillot wrote:
> This looks good to me, and is a good idea.
> 
> Unfortunately, it doesn't really help us in pacemaker, since we have to
> stay compatible with older libqb versions. We'll have to take all these
> formatters out and cast to long long instead. (That seems better than
> defining some ugly constants for format and cast depending on libqb
> support.)
>

Thanks. In that case I'll leave this in git until the next minor release
of libqb.

Chrissie

**Pull request closed without merging**